src/main/scala/com/databricks/labs/automl/sanitize/FeatureCorrelationDetection.scala

```diff
@@ -54,30 +54,34 @@ class FeatureCorrelationDetection(data: DataFrame, fieldListing: Array[String])
   private def computeFeatureCorrelation(): Array[FeatureCorrelationStats] = {
 
     val correlationInteractions = new ArrayBuffer[FeatureCorrelationStats]
-    val redundantRecursionEliminator = new ArrayBuffer[String]
-
+   
     val taskSupport = new ForkJoinTaskSupport(new ForkJoinPool(_parallelism))
     val fieldListingPar = fieldListing.par
     fieldListingPar.tasksupport = taskSupport
 
-
-    fieldListingPar.foreach{ x =>
-      val leftFields = fieldListing.filterNot(_.contains(x)).filterNot(f => redundantRecursionEliminator.contains(f))
-      leftFields.foreach{y =>
+   //In the previous code, both of the features having higher negative or positive correlation were getting removed
+   //because for every pair of feature a and b, in correlationInteraction (a,b) as well as (b,a) was getting added
+    fieldListingPar.foreach(cfeature =>
+    {
+      var currIndex = fieldListingPar.indexOf(cfeature)
```

Comment:
How about we just do a .zipWithIndex and then do an async .foreach over the index positions, iterating over current idx -> idx of par collection size/length to get the calculation for the correlation stat to commit to the Buffer?